app/components/festivals/reservations/full-table-panel.tsx

```
@@ -1,40 +1,68 @@
+   */
+  creditsEnabled: boolean;
 }) {
   const router = useRouter();
   const [pending, startTransition] = useTransition();
-  const [dismissed, setDismissed] = useState(false);
+  const [hidden, setHidden] = useState(false);
+
+  // Read after mount: the server has no way to know what this browser dismissed,
+  // and reading during render would make the markup disagree with hydration.
+  useEffect(() => {
+    setHidden(isFullTableBannerDismissed(festivalId));
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/app/components/festivals/reservations/full-table-panel.tsx:55:5
  53 |   // and reading during render would make the markup disagree with hydration.
  54 |   useEffect(() => {
> 55 |     setHidden(isFullTableBannerDismissed(festivalId));
     |     ^^^^^^^^^ Avoid calling setState() directly within an effect
  56 |   }, [festivalId]);
  57 |
  58 |   if (!offer.offered) return null;

(react-hooks/set-state-in-effect)